src/vulkan/pipeline.cc

```diff
@@ -172,6 +172,45 @@ Result Pipeline::CreateDescriptorSets() {
   return {};
 }
 
+VkPushConstantRange Pipeline::GetPushConstantRange() {
+  if (push_constant_data_.empty())
+    return {};
+
+  auto it =
+      std::min_element(push_constant_data_.begin(), push_constant_data_.end(),
+                       [](const BufferData& a, const BufferData& b) {
+                         return a.offset < b.offset;
+                       });
+  if (it == push_constant_data_.end())
+    return {};
```

Comment:
If I understand std::min_element correctly, this case should never happen.   It seems harmless to keep, except it's slightly confusing.  Maybe I would add an assert here that it isn't the end iterator.

Reply:
done